crates/oxc_linter/src/rules/eslint/no_loss_of_precision.rs

```diff
@@ -48,18 +49,89 @@ impl Rule for NoLossOfPrecision {
     }
 }
 
-#[derive(Debug, Eq)]
-pub struct NormalizedNum<'a> {
-    magnitude: isize,
-    coefficient: Cow<'a, str>,
+pub struct RawNum<'a> {
+    int: &'a str,
+    frac: &'a str,
+    exp: isize,
 }
 
-impl<'a> PartialEq for NormalizedNum<'a> {
+#[derive(Debug)]
+pub struct ScientificNotation<'a> {
+    int: &'a str,
+    frac: Cow<'a, str>,
+    exp: isize,
+    scientific: bool,
+    precision: usize,
+}
+
+impl PartialEq for ScientificNotation<'_> {
     fn eq(&self, other: &Self) -> bool {
-        if self.coefficient == "0" {
-            true
+        if self.int == other.int && self.frac == other.frac {
+            if self.int == "0" && self.frac == "" {
+                return true;
+            }
+            return self.exp == other.exp;
+        }
+        false
+    }
+}
+
+impl<'a> RawNum<'a> {
+    fn new(num: &str) -> Option<RawNum<'_>> {
+        let re = Regex::new(
```

Comment:
Need to move this to a static, otherwise it's constructed every time you run it.